alpine-nix/Dockerfile

```diff
@@ -0,0 +1,47 @@
+# dkubb/alpine-nix
+
+FROM alpine:3.2
+MAINTAINER Dan Kubb <dkubb@fastmail.com>
+
+ENV NIX_VERSION 1.9
+
+# Upgrade installed system dependencies
+COPY apk-packages /tmp/
+RUN sed 's/#.*$//;/^$/d' /tmp/apk-packages | tr -d ' ' | xargs apk add --update-cache
+
+ENV CC       x86_64-alpine-linux-musl-gcc
+ENV CXX      x86_64-alpine-linux-musl-g++
+ENV CFLAGS   -static
+ENV CXXFLAGS -static
+
+# Install WWW::Curl perl module (until perl-www-curl package is available)
+RUN perl -MCPAN -e 'install WWW::Curl'
+
+# Add patch to fix build errors
+COPY nix.patch /tmp/
+
+# Install nix from source
+RUN curl --location "https://github.com/NixOS/nix/archive/$NIX_VERSION.tar.gz" | tar xz -C /tmp \
```

Comment:
There is no validation for this file downloaded being done. Should we look for verifying its contents via a GPG signature? The nice thing is that you only have to trust the key once and all following packages can be updated without maintaining an SHA2.

If GPG is not an option we should validate the SHA2 we manually carry around, to prevent spoofing of the contents.